example/image-classification/benchmark_score.py

```diff
@@ -39,21 +39,32 @@ def get_symbol(network, batch_size):
     net = import_module('symbols.'+network)
     sym = net.get_symbol(num_classes = 1000,
                          image_shape = ','.join([str(i) for i in image_shape]),
-                         num_layers  = num_layers)
+                         num_layers  = num_layers,
+                         dtype=dtype)
     return (sym, [('data', (batch_size,)+image_shape)])
 
-def score(network, dev, batch_size, num_batches):
+def score(network, dev, batch_size, num_batches, dtype):
     # get mod
-    sym, data_shape = get_symbol(network, batch_size)
+    sym, data_shape = get_symbol(network, batch_size, dtype)
     mod = mx.mod.Module(symbol=sym, context=dev)
     mod.bind(for_training     = False,
              inputs_need_grad = False,
              data_shapes      = data_shape)
     mod.init_params(initializer=mx.init.Xavier(magnitude=2.))
 
     # get data
-    data = [mx.random.uniform(-1.0, 1.0, shape=shape, ctx=dev) for _, shape in mod.data_shapes]
+    data = [mx.random.uniform(-1.0, 1.0, shape=shape, ctx=dev, dtype=dtype) for _, shape in mod.data_shapes]
     batch = mx.io.DataBatch(data, []) # empty label
+    mod.forward(batch, is_train=False)
+    if (dtype == 'float16'):
```

Comment:
remove parentheses

Reply:
Also, seems that when binding, the parameters will automatically take the first input's data type. https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/module/module.py#L443